src/hooks/useFocusState.ts

```diff
@@ -27,4 +33,36 @@ const useFocusState = (
   return { isFocused, focusProps };
 };
 
-export { useFocusState };
+const useFocusWithinState = (
+  props: FocusWithinProps
+): { isFocusedWithin: boolean; focusWithinProps: HTMLAttributes<HTMLElement> } => {
+  const [isFocusedWithin, setFocusWithin] = useState(false);
+  const { focusWithinProps } = useFocusWithin({
+    onFocusWithin: (...args) => {
+      setFocusWithin(true);
+      if (props.onFocusWithin) {
+        props.onFocusWithin.apply(this, args);
+      }
```

Comment:
nitpicking: this can be simplified to:
```suggestion
    props.onFocusWithin?.(...args);
```
because the arrow functions `this` is `undefined` anyway

Reply:
I've changed this